generated/csharp/GitHub/Client/OctokitClientFactory.cs

```diff
@@ -0,0 +1,53 @@
+using System.Net;
+using GitHub.Octokit.Client.Middleware;
+
+namespace GitHub.Client;
+
+
+public static class OctokitClientFactory
+{
+    public static HttpClient Create(HttpMessageHandler? finalHandler = null)
+    {
+      var defaultHandlers = CreateDefaultHandlers();
+      var handler = ChainHandlersCollectionAndGetFirstLink(finalHandler ?? GetDefaultHttpMessageHandler(), defaultHandlers.ToArray());
+      return handler != null ? new HttpClient(handler) : new HttpClient();
+    }
+
+    public static IList<DelegatingHandler> CreateDefaultHandlers()
+    {
+        return new List<DelegatingHandler>
+        {
+            new VersionHandler(),
+            new UserAgentHandler(),
+        };
+    }
+
+    public static DelegatingHandler? ChainHandlersCollectionAndGetFirstLink(HttpMessageHandler? finalHandler, params DelegatingHandler[] handlers)
+    {
+      if(handlers == null || !handlers.Any()) return default;
+      var handlersCount = handlers.Length;
+      for(var i = 0; i < handlersCount; i++)
+      {
+          var handler = handlers[i];
+          var previousItemIndex = i - 1;
+          if(previousItemIndex >= 0)
+          {
+              var previousHandler = handlers[previousItemIndex];
+              previousHandler.InnerHandler = handler;
+          }
+      }
+      if(finalHandler != null)
+          handlers[handlers.Length-1].InnerHandler = finalHandler;
+      return handlers.First();
+    }
+
+    public static DelegatingHandler? ChainHandlersCollectionAndGetFirstLink(params DelegatingHandler[] handlers)
+    {
+      return ChainHandlersCollectionAndGetFirstLink(null,handlers);
+    }
+
+    public static HttpMessageHandler GetDefaultHttpMessageHandler(IWebProxy? proxy = null)
+    {
+      return new HttpClientHandler { Proxy = proxy, AllowAutoRedirect = false };
```

Comment:
What's the line of thinking behind AllowAutoRedirect defaulting to false? There are a few requests that can autoredirect validly: for example, when downloading an asset, permissions are checked before redirecting to the asset itself.

Reply:
So this is a really thin implementation of instantiating a Http message handler. The `HttpClientHandler` class has a bunch of other properties that can be set to configure its behavior, such as `UseCookies`, `UseDefaultCredentials`, `MaxAutomaticRedirections`, and `AutomaticDecompression`, among others. However, in this case, only Proxy and AllowAutoRedirect are being set - I wanted to limit the scope of this iteration. 

Reason being just about every property in the `HttpClientHandler` has restrictive decorations that detail which platforms are not supported for the given property. I am pretty sure I was inspired by the kiota code to go down this initial route.  

That being said, this is simply a little factory to get a default handler and is not used if you want to pass over your own `HttpClientHandler` - setting the basics for the default and allowing for the more complex later.

Reply:
Got it; let's keep talking about our default implementation as this evolves!